clang/lib/DPCT/PatternRewriter.cpp

```diff
@@ -37,6 +37,97 @@ struct MatchResult {
   std::unordered_map<std::string, std::string> Bindings;
 };
 
+extern llvm::cl::opt<bool> MigrateCmakeScriptOnly;
```

Comment:
Please  put the logic into separate file which is specific to cmake build scripts migration.

Reply:
the option imported here is used to differentiate the logic specific for cmake script migration in pattern rewriter,  it is better to keep it in my opinion. will change it later if necessary in future.